tools/bin/mbedtls-run-tests

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+
+"""Run Mbed TLS unit tests.
+
+By default, run all test casess. If at least one of --prefix, --regex or
+--substring is specified, only run the test cases that match any of these
+options.
+"""
+
+import argparse
+import os
+import re
+import subprocess
+import tempfile
+
+
+class TestCaseFilter:
+    """Test case filter."""
+
+    def __init__(self, options):
+        """Set up a test case filter.
+
+        See `main` for what options are valid.
+        """
+        include = ''
+        for string in options.substring:
+            include += '|.*' + re.escape(string)
+        for string in options.prefix:
+            include += '|' + re.escape(string)
+        for regex in options.regex:
+            include += '|.*' + regex
+        if include: # no filter specified
+            include = include[1:] # remove the leading '|'
+        self.include = re.compile(include.encode())
```

Comment:
Is calling encode() needed here? I'm not sure why it would be.

Reply:
`include.encode()` is a `bytes`, so `self.include` is a regex matching bytes, which is necessary because `filter_test_cases` opens the file in binary, which avoids barfing on non-ASCII characters if the ambient locale doesn't match the file's locale (at least with older Python versions that don't assume UTF-8).

Types would make this clearer but I haven't gotten into the habit of using types for low-maintenance productivity scripts. I think we should keep the tools in `mbedtls-docs` low-maintenance and not require things like mypy and pylint. If we decide the tool is important enough to warrant that, we should move it to the mbedtls repo.